test/integration/hasMany.test.js

```diff
@@ -265,6 +266,20 @@ describe('hasMany', function () {
         limit: 1
       }]);
     });
+
+    it('should skip batching if include is set', async function() {
+      let DataLoaderSpy = this.sandbox.spy(DataLoader);
+      let project1 = await this.Project.findById(this.project1.id, { include: [ this.Project.associations.members ]});
+      let project2 = await this.Project.findById(this.project2.id, { include: [ this.Project.associations.members ]});
+
+      expect(project1.members, 'not to be undefined');
+      expect(project2.members, 'not to be undefined');
+      expect(project1.members, 'to have length', 3);
+      expect(project2.members, 'to have length', 4);
+
+      expect(DataLoaderSpy.calledWithNew(), 'to be false');
```

Comment:
While i understand the intent i think it's better to simply spy on the load function, but i'll defer to @janmeier here